### ISSUE TITLE
General UI/UX improvement of data table in Table View

### ISSUE THREAD

**Issue by Ig-Rebollo** [kobotoolbox/kpi#3198]
This issue outlines some important improvements to the table view UI. These improvements are divided in two groups: accessibility improvements, which should be done before any other table view related additions; and usability improvements, which can be done now, or after other more urgent additions, depending on timelines.

**Comment by Ig-Rebollo** [comment]
@magicznyleszek @duvld I've been thinking about it, and if we end up doing the monospace font for numbers (remember this one was optional, so if this is something more than a quick and easy change, is not worth wasting time on it right now) let's try using [Roboto mono](https://fonts.google.com/specimen/Roboto+Mono#standard-styles) for the time being. It will align with the rest of the platform better.
